src/peft/utils/other.py

```diff
@@ -227,17 +227,25 @@ def active_adapter(self) -> str:
         # use a property to ensure that active_adapter is not set directly, instead use the set_adapter method
         return self._active_adapter
 
-    @property
-    def weight(self):
-        if self.active_adapter not in self.modules_to_save:
-            return self.original_module.weight
-        return self.modules_to_save[self.active_adapter].weight
-
-    @property
-    def bias(self):
-        if self.active_adapter not in self.modules_to_save:
-            return self.original_module.bias
-        return self.modules_to_save[self.active_adapter].bias
+    def __getattr__(self, name: str):
+        # Note: This whole method may seem overly complex at first but PyTorch messes with __getattr__ in a way that
+        # requires very careful handling to avoid infinite recursion.
+        try:
+            return super().__getattr__(name)
+        except AttributeError:
+            pass
+
+        if "_modules" not in self.__dict__:
+            raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
+
+        # Could not find the attribute the PyTorch way. So let's check if it's an attribute on the
+        # original_module/modules_to_save.
+        modules = self.__dict__["_modules"]
+        if self.disable_adapters:
+            module = modules["original_module"]
+        else:
+            module = modules["modules_to_save"][self.active_adapter]
```

Comment:
That's probably the case but are we sure that `self.active_adapter` is in `modules["modules_to_save"]` since we changed a bit the logic.

Reply:
Yes, normally, there should always be an entry corresponding to `self.active_adapter`. If not, something must have seriously gone wrong (like users overriding private attributes). Still, I have extended the check and added a test just to be on the safe side.

The previous behavior was incorrect, as it didn't check for disabled adapters, this new behavior should be the correct one.